### PR TITLE
Do not allow uploading private fields.

### PR DESCRIPTION
**Pull request by n-rook**
(Also, changes the name of "TAS" to "Tool-Assisted" replays.) #184

### CODE REVIEW

File: project/thscoreboard/replays/models.py [hunk 1]
```diff
@@ -100,7 +100,11 @@ class Category(models.IntegerChoices):
     """
 
     PRIVATE = 4, pgettext_lazy('Category', 'Private')
-    """A private replay that isn't shown to anyone."""
+    """A private replay that isn't shown to anyone.
+
+    Private is deprecated; it will be removed soon. (If someone didn't
```
Comment (raviddog) [inline project/thscoreboard/replays/models.py:105]: We're in full "private" development mode without any users or external tools relying on our code, why do we need to depreciate stuff before taking it out?

**Review by raviddog — COMMENTED** [review]
See comment on issue

File: project/thscoreboard/replays/models.py (hunk 1 above)
Comment (n-rook) [reply project/thscoreboard/replays/models.py:105]: We have beta users now; what if one of them adds a private replay after I submit this PR but before it is released to production? This is unlikely, but it's not totally impossible, and it's best to get in the habit of being safe early (IMO)

**Comment by n-rook** [comment]
yeah I won't merge this until we decide whether we really want to do it

**Review by raviddog — APPROVED** [review]

**Comment by 32th-System** [comment]
so, can we merge this now? The changes are approved

**Comment by n-rook** [comment]
oh yeah we probably should, I've just been lazy. I'll pull in new changes and make sure it still works first, but after that I'll submit it

**Pull request merged**